File: Week 2 Arrays/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    // Prompt user for input text
    string text = get_string("Text: ");

    // Initialize variables to track number of letters, words, and sentences
    int letters = 0, words = 1, sentences = 0;

    // Iterate over every character in the input text
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        // If the character is alphabetic, increment the letters count
        if (isalpha(text[i]))
        {
            letters++;
        }
        // If the character is a space, increment the words count
        else if (isspace(text[i]))
        {
            words++;
        }
        // If the character is a period, exclamation point, or question mark, increment the sentences count
        else if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentences++;
        }
    }

    // Calculate the average number of letters per 100 words and the average number of sentences per 100 words
    float L = (float) letters / words * 100;
    float S = (float) sentences / words * 100;

    // Compute the Coleman-Liau index rounded to the nearest integer
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    // Output the reading level based on the computed index
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: Week 6 Python/sentimental-readability/readability.py
from re import findall


# Get the text input from the user
text = input("Text: ")


# Count the number of letters, words, and sentences in the text
letters = len(findall("[a-zA-Z]", text))
words = len(text.split())
sentences = len(findall("[.!?]", text))


# Calculate the Coleman-Liau index
L = (letters / words) * 100
S = (sentences / words) * 100
index = 0.0588 * L - 0.296 * S - 15.8


# Print the grade level
if index >= 16:
    print("Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print(f"Grade {round(index)}")